src/main/scala/advent/solutions/Day2.scala

```diff
@@ -1,42 +1,95 @@
 package advent.solutions
 
+
+import scala.annotation.tailrec
+
 /** Day 2: 1202 Program Alarm
-  *
-  * @see https://adventofcode.com/2019/day/2
-  */
+ *
+ * @see https://adventofcode.com/2019/day/2
+ */
 object Day2 {
 
   object Part1 {
 
     /** Runs an Intcode program
-      *
-      * @param program A list of opcodes and positions
-      * @return The program after having been run on itself
-      */
-    def run(program: List[Int]): List[Int] = ???
+     *
+     * @param program A list of opcodes and positions
+     * @return The program after having been run on itself
+     */
+    def run(program: List[Int]): List[Int] = {
+      val len = program.length - 1
+
+      @tailrec
+      def go(accum: List[Int], num: Int): List[Int] = {
+        if (num > len) accum else go(repair(num, accum), num + 4)
+      }
+
+      go(program, 0)
+    }
+
+    private def repair(initial: Int, program: List[Int]): List[Int] = {
+      if (program(initial) == 1) repairProgram(program, decodeOne(initial, program))
+      else if (program(initial) == 2) repairProgram(program, decodeTwo(initial, program))
+      else program
+    }
+
+    private def decodeOne(initial: Int, program: List[Int]): (Int, Int) = {
```

Comment:
The values 1 and 2 seem fairly arbitrary, so I probably wouldn't base our function names off of them.  This function:
 - looks up the index that a value should be stored at
 - calculates a value by adding two others together

Instead of calling this `decodeOne`, perhaps we could call it `lookupAndAdd`?  It might even be cleaner to structure it such that it only adds the numbers together:

```scala
 private def lookupAndAdd(initial: Int, program: List[Int]): Int = {
      program(program(initial + 1)) + program(program(initial + 2))
    }
```